lib/objects/objectsUtils.js

```diff
@@ -1,698 +1,698 @@
+    '.md':      {type: 'text/markdown',             binary: false},
+    '.xml':     {type: 'text/xml',                  binary: false},
+    '.svg':     {type: 'image/svg+xml',              binary: false},
+    '.eot':     {type: 'application/vnd.ms-fontobject', binary: true},
+    '.ttf':     {type: 'application/font-sfnt',     binary: true},
+    '.cur':     {type: 'application/x-win-bitmap',  binary: true},
+    '.woff':    {type: 'application/font-woff',     binary: true},
+    '.wav':     {type: 'audio/wav',                 binary: true},
+    '.mp3':     {type: 'audio/mpeg3',               binary: true},
+    '.avi':     {type: 'video/avi',                 binary: true},
+    '.qt':      {type: 'video/quicktime',           binary: true},
+    '.ppt':     {type: 'application/vnd.ms-powerpoint', binary: true},
+    '.pptx':    {type: 'application/vnd.ms-powerpoint', binary: true},
+    '.doc':     {type: 'application/msword',        binary: true},
+    '.docx':    {type: 'application/msword',        binary: true},
+    '.xls':     {type: 'application/vnd.ms-excel',  binary: true},
+    '.xlsx':    {type: 'application/vnd.ms-excel',  binary: true},
+    '.mp4':     {type: 'video/mp4',                 binary: true},
+    '.mkv':     {type: 'video/mkv',                 binary: true},
+    '.zip':     {type: 'application/zip',           binary: true},
+    '.ogg':     {type: 'audio/ogg',                 binary: true},
+    '.manifest':{type: 'text/cache-manifest',       binary: false},
+    '.pdf':     {type: 'application/pdf',           binary: true},
+    '.gz':      {type: 'application/gzip',          binary: true},
+    '.gzip':    {type: 'application/gzip',          binary: true}
+};
+
+let users                = {};
+let groups               = {};
+
+function getMimeType(ext) {
+    if (!ext) {
+        return {mimeType: 'text/html', isBinary: false};
+    }
+    if (ext instanceof Array) ext = ext[0];
+    ext = ext.toLowerCase();
+    let _mimeType = 'text/javascript';
+    let isBinary  = false;
+
+    if (mimeTypes[ext]) {
+        _mimeType = mimeTypes[ext].type;
+        isBinary = mimeTypes[ext].binary;
+    }
+
+    return {mimeType: _mimeType, isBinary: isBinary};
+}
+
+/**
+ * @class
+ * Writable memory stream
+ */
+function WMStrm(key, options) {
+    // allow use without new operator
+    if (!(this instanceof WMStrm)) return new WMStrm(key, options);
+
+    Writable.call(this, options); // init super
+    this.key = key; // save key
+    memStore[key] = Buffer.alloc(0); // empty
+}
+util.inherits(WMStrm, Writable);
+
+WMStrm.prototype._write = function (chunk, enc, cb) {
+    if (chunk) {
+        // our memory store stores things in buffers
+        const buffer = (Buffer.isBuffer(chunk)) ?
+            chunk :  // already is Buffer use it
+            Buffer.from(chunk, enc);  // string, convert
+
+        // concatenate to the buffer already there
+        if (!memStore[this.key]) {
```

Comment:
Does this even happen if we're initializing `memStore[key]` in line 124 with an empty buffer?

Reply:
Don't know Code from BF :-)
Bug in fact in L161 there is a delete memStore[...] ... so theoretically could

Reply:
Ideally we would have tests that tell us :see_no_evil: